io_scene_niftools/modules/nif_import/animation/transform.py

```diff
@@ -49,6 +50,47 @@
 from io_scene_niftools.utils import math
 from io_scene_niftools.utils.blocks import safe_decode
 from io_scene_niftools.utils.logging import NifLog
+from io_scene_niftools.utils.consts import QUAT, EULER, LOC, SCALE
+
+
+def as_b_quat(n_val):
+    return mathutils.Quaternion([n_val.w, n_val.x, n_val.y, n_val.z])
+
+
+def as_b_loc(n_val):
+    return mathutils.Vector([n_val.x, n_val.y, n_val.z])
+
+
+def as_b_scale(n_val):
+    return n_val, n_val, n_val
+
+
+def as_b_euler(n_val):
+    return mathutils.Euler(n_val)
+
+
+def correct_loc(key, n_bind_rot_inv, n_bind_trans):
+    return math.import_keymat(n_bind_rot_inv, mathutils.Matrix.Translation(key - n_bind_trans)).to_translation()
+
+
+def correct_quat(key, n_bind_rot_inv, n_bind_trans):
+    return math.import_keymat(n_bind_rot_inv, key.to_matrix().to_4x4()).to_quaternion()
+
+
+def correct_euler(key, n_bind_rot_inv, n_bind_trans):
+    return math.import_keymat(n_bind_rot_inv, key.to_matrix().to_4x4()).to_euler()
+
+
+def correct_scale(key, n_bind_rot_inv, n_bind_trans):
```

Comment:
hmmm...
Suspect this is not needed/used

Reply:
Yes it is here in the lookup table:
key_lut = {
    QUAT: (as_b_quat, correct_quat, 4),
    EULER: (as_b_euler, correct_euler, 3),
    LOC: (as_b_loc, correct_loc, 3),
    SCALE: (as_b_scale, correct_scale, 3),
}